minio/credentials/providers.py

```diff
@@ -196,37 +196,38 @@ class EnvAWSProvider(Provider):
     """Credential provider from AWS environment variables."""
 
     def __init__(self):
-        access_key = (
+        self._access_key = (
             os.environ.get("AWS_ACCESS_KEY_ID") or
             os.environ.get("AWS_ACCESS_KEY")
         )
-        secret_key = (
+        self._secret_key = (
             os.environ.get("AWS_SECRET_ACCESS_KEY") or
             os.environ.get("AWS_SECRET_KEY")
         )
-        self._credentials = Credentials(
-            access_key,
-            secret_key,
-            session_token=os.environ.get("AWS_SESSION_TOKEN"),
-        )
+        self._session_token = os.environ.get("AWS_SESSION_TOKEN")
 
     def retrieve(self):
         """Retrieve credentials."""
-        return self._credentials
+        return Credentials(
+            access_key=self._access_key,
+            secret_key=self._secret_key,
+            session_token=self._session_token,
+        )
```

Comment:
As `os.environ` is modifiable at runtime, just fetch access/secret keys and session token every time. There is no need to store them in `__init__()`

Reply:
I just kept the same logic, that is, look for env vars at object creation and don't change them anymore. If we now fetch, and possibly change, them every time we may break something already running... ? Just thinking.